dip-0004.md

```diff
@@ -132,6 +165,23 @@ To achieve this, It is suggested to first request a full masternode list from an
   6. Calculate the hash of “cbTx” and verify existence of this transaction in the block specified by “blockHash”. To do this, use the already received block header and the fields “totalTransactions”, “merkleHashes” and “merkleFlags” from the `MNLISTDIFF` message and perform a merkle verification the same way as done when a “MERKLEBLOCK” message is received. If the verification fails, abort the process and ask for diffs from another node.
   7. Store the resulting validated masternode list identified by “blockHash”
 
+## Tracking/Updating and verifying the active LLMQ sets based on MNLISTDIFF
+
+With the introduction of version 2 CbTx special transactions, a merkle root for the currently active LLMQ sets was added
+to the coinbase. This merkle root allows (SPV) clients to maintain the active LLMQ sets without the full chain.
+
+It is suggested to couple the house-/bookkeeping of active LLMQ sets with the masternode list bookkeeping described in
+the previous section and perform the following additional steps:
+
+  1. Create a copy of the active LLMQ sets which were given at "baseBlockHash". If “baseBlockHash” is all-zero, empty sets must be used.
+  2. Delete all entries found in "deletedQuorums" from the corresponding active LLMQ sets.
+  3. Verify each final commitment found in "newQuorums", by the same rules found in [DIP6 - Long-Living Masternode Quorums](https://github.com/dashpay/dips/blob/master/dip-0006.md).
```

Comment:
This step should be dropped. SPVs shouldn't verify this, instead they should rely on PoW+merkleroot (steps 5 and 6) - merkle root won't match if some final commitment was tampered.